src/main/java/com/conveyal/r5/analyst/scenario/RemoveStops.java

```diff
@@ -147,23 +158,63 @@ public TripPattern processTripPattern (TripPattern originalTripPattern) {
             int accumulatedRideTime = 0;
             int prevInputDeparture = 0;
             int prevOutputDeparture = 0;
+            int nStopsRemovedSinceLastStop = 0;
             for (int i = 0, j = 0; i < removeStop.length; i++) {
                 int rideTime = originalSchedule.arrivals[i] - prevInputDeparture;
                 int dwellTime = originalSchedule.departures[i] - originalSchedule.arrivals[i];
                 prevInputDeparture = originalSchedule.departures[i];
                 if (removeStop[i]) {
-                    // The current stop will not be included in the output. Record the travel time.
+                    // The current stop will not be included in the output. Record the travel time and subtract off the
+                    // seconds saved. This may cause accumulatedRideTime to go negative, which may be fine since we're putting
+                    // all the time savings before the stop. After we've accumulated ride time for a whole segment,
+                    // we check to make sure it's positive.
                     accumulatedRideTime += rideTime;
+                    nStopsRemovedSinceLastStop++;
```

Comment:
It seems odd to me to count how many stops have been removed and then multiply that by the number of stops removed. Shouldn't it be possible to subtract secondsSavedAtEachStop out of accumulatedRideTime at each iteration, and just keep an eye on the accumulated total?